src/lib.rs

```diff
@@ -207,25 +207,37 @@ pub fn run(
                 // This is only to keep the context for tests
                 Variant::Ret(ret_variant) => match ret_variant {
                     RetOpcode::Ok => {
-                        let should_break = ok(&mut vm, &opcode)?;
-                        if should_break {
-                            break;
+                        let should_break = ok(&mut vm, &opcode);
+                        if should_break.is_err() {
+                            Err(should_break.unwrap_err())
+                        } else {
+                            if should_break.unwrap() {
+                                break;
+                            }
+                            Ok(())
                         }
-                        Ok(())
                     }
                     RetOpcode::Revert => {
-                        let should_break = revert(&mut vm, &opcode)?;
-                        if should_break {
-                            return Ok((ExecutionOutput::Revert(vec![]), vm));
+                        let should_break = revert(&mut vm, &opcode);
+                        if should_break.is_err() {
+                            Err(should_break.unwrap_err())
+                        } else {
+                            if should_break.unwrap() {
+                                return Ok((ExecutionOutput::Revert(vec![]), vm));
+                            }
+                            Ok(())
                         }
-                        Ok(())
                     }
                     RetOpcode::Panic => {
-                        let should_break = panic(&mut vm, &opcode)?;
-                        if should_break {
-                            return Ok((ExecutionOutput::Panic, vm));
+                        let should_break = panic(&mut vm, &opcode);
+                        if should_break.is_err() {
+                            Err(should_break.unwrap_err())
+                        } else {
+                            if should_break.unwrap() {
+                                return Ok((ExecutionOutput::Panic, vm));
+                            }
+                            Ok(())
                         }
-                        Ok(())
                     }
                 },
```

Comment:
Simplify by `match`ing:
```rs
                Variant::Ret(ret_variant) => match ret_variant {
                    RetOpcode::Ok => match ok(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                break;
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                    RetOpcode::Revert => match revert(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                return Ok((ExecutionOutput::Revert(vec![]), vm));
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                    RetOpcode::Panic => match panic(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                return Ok((ExecutionOutput::Panic, vm));
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                },
```

Reply:
Even simpler (maybe not that legible?):
```rs
                Variant::Ret(ret_variant) => {
                    let result = match ret_variant {
                        RetOpcode::Ok => ok(&mut vm, &opcode),
                        RetOpcode::Revert => revert(&mut vm, &opcode),
                        RetOpcode::Panic => panic(&mut vm, &opcode),
                    };
                    match result {
                        Ok(should_break) => {
                            if should_break {
                                break;
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    }
                }
```

Reply:
The second one would not work, since the return values are different, I'll do the first one, thanks!

Reply:
You are right, missed it!